### ISSUE TITLE
为什么分类层不使用混合精度进行训练？

### ISSUE THREAD

**Issue by ZHAIXINGZHAIYUE** [deepinsight/insightface#1684]

**Comment by anxiangsir** [comment]
logits 有指数运算，涉及到相加的部分，最终还要cast成fp32，不会有显存的减幅，不过能加速训练。

**Comment by anxiangsir** [comment]
最近又研究了一下，FC层cast成FP16，对显存不会有帮助，但是对速度还是有提升的，因为分类层的内积只有512个fp16的相乘和相加，对精度也不会又影响
过一段时间会把这个更新上来

**Comment by ZHAIXINGZHAIYUE** [comment]
向你学习

**Comment by ZHAIXINGZHAIYUE** [comment]
您好，请教一下，clip_grad_norm使用的时候，其中max_norm值的设置有什么经验吗？

**Comment by anxiangsir** [comment]
一般训tansformer的时候，论文给的超参数都是这个值，clip_grad_norm对resnet没啥用，下一个更新回去掉

**Comment by ZHAIXINGZHAIYUE** [comment]
> 一般训tansformer的时候，论文给的超参数都是这个值，clip_grad_norm对resnet没啥用，下一个更新会去掉

另外，我发现在resnet中用这个，其实会影响权重的分布。相比于mxnet,会让权重更小。
